hrjobcontract/js/src/job-contract/controllers/contract.js

```diff
@@ -68,28 +69,26 @@ define([
                 });
             };
 
-            function updateContractList(newEndDate){
-                var isCurrentContract;
-
-                //Is contract end date changed
-                if ($scope.details.period_end_date ?
-                new Date($scope.details.period_end_date).getTime() !== new Date(newEndDate).getTime() :
-                !!$scope.details.period_end_date !== !!newEndDate) {
-
-                    isCurrentContract = !newEndDate || new Date(newEndDate).setHours(0,0,0,0) >= new Date().setHours(0,0,0,0);
-
-                    if (isCurrentContract != !!+$scope.$parent.contract.is_current) {
-                        if (isCurrentContract) {
-                            $scope.$parent.contract.is_current = '1';
-                            $scope.$parent.$parent.contractCurrent.push($scope.$parent.contract);
-                            $scope.$parent.$parent.contractPast.splice($scope.$parent.$parent.contractPast.indexOf($scope.$parent.contract),1);
-                        } else {
-                            $scope.$parent.contract.is_current = '0';
-                            $scope.$parent.$parent.contractPast.push($scope.$parent.contract);
-                            $scope.$parent.$parent.contractCurrent.splice($scope.$parent.$parent.contractCurrent.indexOf($scope.$parent.contract),1)
-                        }
-                    }
-                }
+            /**
+              * Updates the contract list view,
+              * by sorting the contract into current or past
+              * depending on the period end date of the contract.
+              *
+              *
+              * @param {type} newEndDate the date specified by the user
+            */
+            $scope.updateContractList = function(newEndDate){
```

Comment:
As an internal function testing it would be difficult to test it, the other alternative would be to `this.updateContractList = function(newEndDate){`.